extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeCommentsExtractor.java

```diff
@@ -0,0 +1,266 @@
+                    try {
+                        return c.get("authorThumbnail").get("thumbnails").get(2).get("url").asText();
+                    } catch (Exception e) {
+                        throw new ParsingException("Could not get author thumbnail", e);
+                    }
+                }
+
+                @Override
+                public String getAuthorName() throws ParsingException {
+                    try {
+                        return c.get("authorText").get("simpleText").asText();
+                    } catch (Exception e) {
+                        throw new ParsingException("Could not get author name", e);
+                    }
+                }
+
+                @Override
+                public String getAuthorEndpoint() throws ParsingException {
+                    try {
+                        return "https://youtube.com"
+                                + c.get("authorEndpoint").get("browseEndpoint").get("canonicalBaseUrl").asText();
+                    } catch (Exception e) {
+                        throw new ParsingException("Could not get author endpoint", e);
+                    }
+                }
+            };
+
+            collector.commit(extractor);
+        });
+
+    }
+
+    private void fetchTitle(JsonNode ajaxJson) {
+        if(null == title) {
+            try {
+                title = ajaxJson.findValue("commentTargetTitle").get("simpleText").asText();
+            } catch (Exception e) {
+                title = "Youtube Comments";
+            }
+        }
+    }
+
+    @Override
+    public void onFetchPage(Downloader downloader) throws IOException, ExtractionException {
+        DownloadResponse response = downloader.get(getUrl());
+        String responseBody = response.getResponseBody();
+        cookies = response.getResponseHeaders().get("Set-Cookie");
+        sessionToken = findValue(responseBody, "XSRF_TOKEN");
+        String commentsToken = findValue(responseBody, "COMMENTS_TOKEN");
+        initPage = getPage(getNextPageUrl(commentsToken));
+    }
+
+    @Override
+    public String getName() throws ParsingException {
+        return title;
+    }
+
+    private String makeAjaxRequest(String siteUrl) throws IOException, ReCaptchaException {
+
+        StringBuilder postData = new StringBuilder();
+        postData.append(URLEncoder.encode("session_token", "UTF-8"));
+        postData.append('=');
+        postData.append(URLEncoder.encode(sessionToken, "UTF-8"));
+
+        Map<String, List<String>> requestHeaders = new HashMap<>();
+        requestHeaders.put("Content-Type", Arrays.asList("application/x-www-form-urlencoded"));
+        requestHeaders.put("Accept", Arrays.asList("*/*"));
+        requestHeaders.put("User-Agent", Arrays.asList(USER_AGENT));
+        requestHeaders.put("X-YouTube-Client-Version", Arrays.asList("2.20180815"));
+        requestHeaders.put("X-YouTube-Client-Name", Arrays.asList("1"));
```

Comment:
Can this client name not be found in a script or somewhere else in the resources ?